src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1050,21 +1250,37 @@ protected virtual void AssignImpl(BoundNode node, BoundExpression value, RefKind
                         else
                         {
                             int slot = MakeSlot(local);
-                            SetSlotState(slot, written);
+                            if (IsCapturedInLocalFunction(localSymbol))
+                            {
+                                AssignVarInLocalFunction(slot, localSymbol);
```

Comment:
>AssignVarInLocalFunction [](start = 32, length = 24)

This appears to circumvent flow analysis of the local function to determine if it was written on every control path through the local function.